app/javascript/components/assessment/steps/SymptomsAssessment.js

```diff
@@ -135,7 +178,23 @@ class SymptomsAssessment extends React.Component {
           </div>
         }
         className="pb-2"
-        onChange={this.handleChange}></Form.Check>
+        onChange={this.handleBoolChange}></Form.Check>
+    );
+  };
+
+  integerSymptom = symp => {
+    const key = `key_${symp.name}${this.props.idPre ? '_idpre' + this.props.idPre : ''}`;
+    const id = `${symp.name}${this.props.idPre ? '_idpre' + this.props.idPre : ''}`;
+    return (
+      <Form.Row className="pt-3" key={key}>
+        <Form.Label className="nav-input-label" key={key + '_label'} htmlFor={id}>
+          <b>{this.props.translations[this.props.lang]['symptoms'][symp.name]['name']}</b>{' '}
+          {this.props.translations[this.props.lang]['symptoms'][symp.name]['notes']
+            ? ' ' + this.props.translations[this.props.lang]['symptoms'][symp.name]['notes']
+            : ''}
+        </Form.Label>
+        <Form.Control size="lg" id={id} key={key + '_control'} className="form-square" value={symp.value || ''} maxLength="9" onChange={this.handleIntChange} />
```

Comment:
The only differences between `integerSymptom` and `floatSymptom` are the `maxLength` value and the `onChange` handler on this `Form.Control`. I'd suggest making this one function that takes those as params, something like `intOrFloatSymptom` or whatever, just because otherwise there's a lot of duplication here. If you want you could also just have one function `intOrFloatSymptom` that then calls `integerSymptomInput` or `floatSymptomInput` which will render this `Form.Control` in case there end up being any other differences there in the future. Either approach would work, the latter one might actually be better in the long run.

Reply:
Good call, I will implement your latter suggestion!